src/riak_kv_app.erl

```diff
@@ -49,7 +49,21 @@
 start(_Type, _StartArgs) ->
     riak_core_util:start_app_deps(riak_kv),
 
-    FSM_Limit = app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT),
+    FSM_Limit = case app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT) of
```

Comment:
Please refactor this code into its own function - it's now too much to just embed in another function. Maybe still get the env here, but pass it to a "determine_fsm_limit/1" that has different function heads for each case?